src/main/java/com/ibm/watson/apis/conversation_enhanced/rest/ProxyResource.java

```diff
@@ -143,7 +147,29 @@ private MessageResponse getWatsonResponse(MessageRequest request, String id) thr
         output.put("CEPayload", docs); //$NON-NLS-1$
       }
     }
-    
+
+    // Log User input and output from Watson
+    if (Boolean.TRUE.equals(LOGGING_ENABLED)) {
+      Logging cloudantLogging = new Logging();
```

Comment:
can you extract the contents of this if block to a new method? It's in some of the main code, which is fine, but I'd like like to keep the number of lines dedicated to this in this "main" method as small as possible